src/Strategy/NuGet/PackagesConfig.hs

```diff
@@ -0,0 +1,93 @@
+module Strategy.NuGet.PackagesConfig
+  ( discover
+  , strategy
+  , buildGraph
+  , analyze
+  , parsePackagesConfig
+
+  ) where
+
+import Prologue
+
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import           Polysemy
+import           Polysemy.Error
+import           Polysemy.Output
+import qualified Text.XML.Light as XML
+
+import           Diagnostics
+import           DepTypes
+import           Discovery.Walk
+import           Effect.ReadFS
+import           Graphing (Graphing, unfold)
+import           Types
+
+discover :: Discover
+discover = Discover
+  { discoverName = "packagesconfig"
+  , discoverFunc = discover'
+  }
+
+discover' :: Members '[Embed IO, Output ConfiguredStrategy] r => Path Abs Dir -> Sem r ()
+discover' = walk $ \_ _ files -> do
+  case find (\f -> (fileName f) == "packages.config") files of
+    Just file -> output (configure file)
+    Nothing -> pure ()
+
+  walkContinue
+
+strategy :: Strategy BasicFileOpts
+strategy = Strategy
+  { strategyName = "nuget-packagesconfig"
+  , strategyAnalyze = analyze
+  , strategyModule = parent . targetFile
+  , strategyOptimal = NotOptimal
+  , strategyComplete = NotComplete
+  }
+
+analyze :: Members '[ReadFS, Error ReadFSErr] r => BasicFileOpts -> Sem r (Graphing Dependency)
+analyze BasicFileOpts{..} = do
+      contents <- readContentsBS targetFile
+      let packagesConfig = parsePackagesConfig =<< XML.parseXMLDoc contents
```

Comment:
Can we extract this pattern to a helper in `ReadFS`? similar to json/parsec/yaml. We can use that helper in both of these strategies